src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -293,36 +294,42 @@ public SCM build(@NonNull SCMHead head, @CheckForNull SCMRevision revision) {
         }
         return result;
     }
-    
+
     /**
      * Returns true if the branchName isn't matched by includes or is matched by excludes.
-     * 
+     *
      * @param branchName
      * @return true if branchName is excluded or is not included
      */
     protected boolean isExcluded (String branchName){
       return !Pattern.matches(getPattern(getIncludes()), branchName) || (Pattern.matches(getPattern(getExcludes()), branchName));
     }
-    
+
     /**
-     * Returns the pattern corresponding to the branches containing wildcards. 
-     * 
+     * Returns the pattern corresponding to the branches containing wildcards.
+     *
      * @param branchName
      * @return pattern corresponding to the branches containing wildcards
      */
     private String getPattern(String branches){
       StringBuilder quotedBranches = new StringBuilder();
-      for (String wildcard : branches.split(" ")){
+      for (String term : branches.split(" ")){
         StringBuilder quotedBranch = new StringBuilder();
-        for(String branch : wildcard.split("\\*")){
-          if (wildcard.startsWith("*") || quotedBranches.length()>0) {
-            quotedBranch.append(".*");
-          }
-          quotedBranch.append(Pattern.quote(branch));
-        }
-        if (wildcard.endsWith("*")){
-          quotedBranch.append(".*");
+
+        if (term.startsWith("^")) { // Regex mode, ^ is not a valid character in git names
```

Comment:
This change (both as it stands, and as amended to use the ':' delimiter) will preserve compatibility with the bug, in fact, I even added a test case to confirm that.